### ISSUE TITLE
AttributeError: 'Attribute' object has not attribute 'name'

### ISSUE THREAD

**Issue by learnitall** [PyCQA/pylint#4680]
Seems to be an issued directly related to PR https://github.com/PyCQA/pylint/pull/3998 and issue https://github.com/PyCQA/pylint/issues/3742. When using a metaclass from a subpackage that can't be imported, and then referencing another variable from said subpackage within the same scope, an ``AttributeError`` is raised. Removing the ``assert`` line in the given file below prevents the error. Came across this while writing unit tests, hence the ``assert`` statement.

### Steps to reproduce

Given a file `lintme.py`:
```python
# pylint: disable=C0304,C0114,C0116,C0115,R0903
import pkg.sub

class Failed(metaclass=pkg.sub.Metaclass):
    pass
assert pkg.sub.value is None
```

### Current behavior

Result of `pylint lintme.py`:
```
(pylint-bug) [rydrew@redhat-worklp pylint-error]$  pylint lintme.py
************* Module lintme
lintme.py:2:0: E0401: Unable to import 'pkg.sub' (import-error)
Exception on node <FunctionDef.test l.4 at 0x7f867ba766a0> in file '/home/rydrew/Documents/pylint-error/lintme.py'
Traceback (most recent call last):
  File "/home/rydrew/.pyenv/versions/pylint-bug/bin/pylint", line 8, in <module>
    sys.exit(run_pylint())
  File "/home/rydrew/.pyenv/versions/pylint-bug/lib64/python3.9/site-packages/pylint/__init__.py", line 24, in run_pylint
    PylintRun(sys.argv[1:])
  File "/home/rydrew/.pyenv/versions/pylint-bug/lib64/python3.9/site-packages/pylint/lint/run.py", line 384, in __init__
    linter.check(args)
  File "/home/rydrew/.pyenv/versions/pylint-bug/lib64/python3.9/site-packages/pylint/lint/pylinter.py", line 973, in check
    self._check_files(
  File "/home/rydrew/.pyenv/versions/pylint-bug/lib64/python3.9/site-packages/pylint/lint/pylinter.py", line 1007, in _check_files
    self._check_file(get_ast, check_astroid_module, name, filepath, modname)
  File "/home/rydrew/.pyenv/versions/pylint-bug/lib64/python3.9/site-packages/pylint/lint/pylinter.py", line 1033, in _check_file
    check_astroid_module(ast_node)
  File "/home/rydrew/.pyenv/versions/pylint-bug/lib64/python3.9/site-packages/pylint/lint/pylinter.py", line 1170, in check_astroid_module
    retval = self._check_astroid_module(
  File "/home/rydrew/.pyenv/versions/pylint-bug/lib64/python3.9/site-packages/pylint/lint/pylinter.py", line 1215, in _check_astroid_module
    walker.walk(ast_node)
  File "/home/rydrew/.pyenv/versions/pylint-bug/lib64/python3.9/site-packages/pylint/utils/ast_walker.py", line 77, in walk
    self.walk(child)
  File "/home/rydrew/.pyenv/versions/pylint-bug/lib64/python3.9/site-packages/pylint/utils/ast_walker.py", line 79, in walk
    callback(astroid)
  File "/home/rydrew/.pyenv/versions/pylint-bug/lib64/python3.9/site-packages/pylint/checkers/variables.py", line 853, in leave_functiondef
    self._check_metaclasses(node)
  File "/home/rydrew/.pyenv/versions/pylint-bug/lib64/python3.9/site-packages/pylint/checkers/variables.py", line 2098, in _check_metaclasses
    consumed.extend(self._check_classdef_metaclasses(child_node, node))
  File "/home/rydrew/.pyenv/versions/pylint-bug/lib64/python3.9/site-packages/pylint/checkers/variables.py", line 2142, in _check_classdef_metaclasses
    name = klass._metaclass.expr.name
AttributeError: 'Attribute' object has no attribute 'name'
```

### Expected behavior

Only the import error is shown

```
(pylint-bug) [rydrew@redhat-worklp pylint-error]$  pylint lintme.py
************* Module lintme
lintme.py:2:0: E0401: Unable to import 'pkg.sub' (import-error)
```

### pylint --version output

Result of `pylint --version` output:

```
pylint 3.0.0-a4
astroid 2.6.2
Python 3.9.5 (default, May 14 2021, 00:00:00) 
[GCC 11.1.1 20210428 (Red Hat 11.1.1-1)]
```

**Comment by Pierre-Sassoulas** [comment]
Thank you for opening the issue, I can reproduce this.
